### PR TITLE
add lint test:-confusing-naming

### PR DESCRIPTION
**Pull request by hellspawn679**
## Which problem is this PR solving?
- (https://github.com/jaegertracing/jaeger/issues/5506)

## Description of the changes
- fixes the `make lint` check that was failing 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
there are no lint rules that are currently failing on the main branch. Where do these come from?

**Comment by hellspawn679** [comment]
> there are no lint rules that are currently failing on the main branch. Where do these come from?

when I run` make lint` locally I was getting these error's 
```
INFO [runner] linters took 727.619655ms with stages: goanalysis_metalinter: 702.156146ms 
cmd/anonymizer/app/anonymizer/anonymizer.go:129:12: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err := os.WriteFile(filepath.Clean(a.mappingFile), dat, os.ModePerm); err != nil {
                  ^
cmd/internal/flags/admin_test.go:59:2: negative-positive: use assert.Positive (testifylint)
        assert.Greater(t, port, 0)
```

**Comment by yurishkuro** [comment]
you may have different version of the tools, try `make install-test-tools`

File: cmd/collector/app/handler/otlp_receiver.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ var _ component.Host = (*otelHost)(nil) // API check
 // StartOTLPReceiver starts OpenTelemetry OTLP receiver listening on gRPC and HTTP ports.
 func StartOTLPReceiver(options *flags.CollectorOptions, logger *zap.Logger, spanProcessor processor.SpanProcessor, tm *tenancy.Manager) (receiver.Traces, error) {
 	otlpFactory := otlpreceiver.NewFactory()
-	return startOTLPReceiver(
+	return start_OTLP_Receiver(
```
Comment (yurishkuro) [inline cmd/collector/app/handler/otlp_receiver.go:47]: This is not what the linter was complaining about. Your names violate Go coding practices.

Comment (hellspawn679) [reply cmd/collector/app/handler/otlp_receiver.go:47]: so how should i fix this naming  problem (Warns on methods with names that differ only by capitalization) ?

Comment (hellspawn679) [reply cmd/collector/app/handler/otlp_receiver.go:47]: will it be okay if i name it like `startOTLPReceiverInternal`

Comment (varshith257) [reply cmd/collector/app/handler/otlp_receiver.go:47]: I think it is fine to use startOTLPReceiver as of camelCase which is generally used in Go.

Comment (yurishkuro) [reply cmd/collector/app/handler/otlp_receiver.go:47]: `startOTLPReceiverInternal` seems ok, but this needs to be decided on a case by case basis, there may be better options in other situations.

**Comment by yurishkuro** [comment]
Most of the changes proposed are much worse than what we have.

**Pull request closed without merging**